docs/design/NsfsVersioning.md

```diff
@@ -49,6 +49,11 @@ In the figure, each original directory contains a hidden .versions/ sub- directo
 * The version ID will be attached as an xattr of the file.
 * When a latest version is moved to past versions directory, the file name will be changed from key to key_{version_id}.
 
+#### Multipart upload version order
+According to AWS specifications, multipart upload version time should be calculated based on multipart upload creation time rather than completion time (see [AWS Documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html#distributedmpupload)).   
+On the other hand, for directory buckets, the object creation time is the completion date of the multipart upload  (see [AWS Documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-express-using-multipart-upload.html#s3-express-distributedmpupload).  
+In our design, the version-id time is calculated based on completion time.
```

Comment:
Could you move this part before the "out of scope" as it is quite an internal and deep subject (and you attached between basic concepts - version ID and Delete Marker)?